### PR TITLE
#27 created object Reservation, ReservationDao and ReservationService

### PR DESCRIPTION
**Pull request by demurelebowski**
#27 CRUD methods for reservation table

### CODE REVIEW

File: src/main/java/javaintern/entity/Reservation.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	private LocalDateTime fromDate;
+	private LocalDateTime toDate;
+	private Double price;
+	private Double quote;
+	private String currency;
+	private Integer guests;
+	private String notes;
+	private Date version;
+
+	public Reservation(Integer id, Integer organizationId, Integer agentId, Integer customerId, Integer productId, ReservationState state) {
```
Comment (vasylnikorych) [inline src/main/java/javaintern/entity/Reservation.java:24]: why these parameters?

File: src/main/java/javaintern/constant/ReservationState.java [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package javaintern.constant;
+
+public enum ReservationState {
+	INITIAL, CONFIRMED, FULLY_PAID, CANCELLED
```
Comment (vasylnikorych) [inline src/main/java/javaintern/constant/ReservationState.java:4]: change style according:
State of record (Initial, Confirmed, FullyPaid, Cancelled)

File: src/main/java/javaintern/entity/Reservation.java (hunk 1 above)
Comment (demurelebowski) [reply src/main/java/javaintern/entity/Reservation.java:24]: I'm not sure what parameters i need here. I'll change it to public Reservation() {}

File: src/main/java/javaintern/dto/ReservationDto.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+	private Integer guests;
+
+	@JsonProperty("notes")
+	private String notes;
+
+	@JsonProperty("version")
+	@JsonSerialize(using = DateSerializer.class)
+	private Date version;
+
+	public ReservationDto(Reservation reservation) {
```
Comment (rdovgan) [inline src/main/java/javaintern/dto/ReservationDto.java:59]: Need to add default constructor

File: src/main/java/javaintern/entity/Reservation.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
+package javaintern.entity;
+
+import javaintern.constant.ReservationState;
+import javaintern.dto.ReservationDto;
+
+import java.time.LocalDateTime;
+import java.util.Date;
+
+public class Reservation {
```
Comment (rdovgan) [inline src/main/java/javaintern/entity/Reservation.java:9]: Add lombok

**Review by rdovgan — CHANGES_REQUESTED** [review]

File: src/main/java/javaintern/constant/ReservationState.java [hunk 5]
```diff
@@ -0,0 +1,5 @@
+package javaintern.constant;
+
+public enum ReservationState {
+	Initial, Confirmed, FullyPaid, Cancelled
```
Comment (DenisTymchuk) [inline src/main/java/javaintern/constant/ReservationState.java:4]: It's better to use UPPERCASE in enum naming

Comment (demurelebowski) [reply src/main/java/javaintern/constant/ReservationState.java:4]: vasylnikorych told me to use CamelCase
vasylnikorych [Nov 4, 2022]
change style according:
State of record (Initial, Confirmed, FullyPaid, Cancelled)

**Pull request merged**